connector/connect/common/src/main/protobuf/spark/connect/relations.proto

```diff
@@ -363,6 +364,23 @@ message Deduplicate {
   optional bool all_columns_as_keys = 3;
 }
 
+// Relation of type [[DeduplicateWithinWatermark]] which have duplicate rows removed within the time
+// range of watermark, could consider either only the subset of columns or all the columns.
+message DeduplicateWithinWatermark {
```

Comment:
Optional: We can just reuse Deduplicate message inside here :).

Reply:
What I can think of is to update to something like below:
```
message DeduplicateWithinWatermark {

  // (Required) Reuse the Deduplicate message for a DeduplicateWithinWatermark.
  Deduplicate deduplicate = 1;
}
```
But if we do that, `DeduplicateWithinWatermark.input` would change to `DeduplicateWithinWatermark.deduplicate.input`. Is there a way to avoid that? If not, I think it's probably better to keep what it is today.

Reply:
Yeah, this option does not look good. We could add a flag 'within_watermark' to Deduplicate message. That way we can reuse the code both on the client and server side.

Reply:
Makes sense, updated.